### ISSUE TITLE
Nikola 7.4 : error building site

### ISSUE THREAD

**Issue by meduz** [getnikola/nikola#1694]
After upgrading to Nikola 7.4, I get the following error:

````
$ nikola build
Scanning posts.Traceback (most recent call last):
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/doit/doit_cmd.py", line 165, in run
    return command.parse_execute(args)
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/doit/cmd_base.py", line 122, in parse_execute
    return self.execute(params, args)
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/doit/cmd_base.py", line 380, in execute
    self, params, args)
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/nikola/__main__.py", line 254, in load_tasks
    self.nikola.gen_tasks('render_site', "Task", 'Group of tasks to render the site.'))
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/doit/loader.py", line 296, in generate_tasks
    for task_dict, x_doc in flat_generator(gen_result, gen_doc):
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/doit/loader.py", line 28, in flat_generator
    for item in gen:
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/nikola/nikola.py", line 1329, in gen_tasks
    for task in flatten(pluginInfo.plugin_object.gen_tasks()):
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/nikola/nikola.py", line 1323, in flatten
    for t in task:
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/nikola/plugins/task/archive.py", line 131, in gen_tasks
    self.site.scan_posts()
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/nikola/nikola.py", line 1419, in scan_posts
    self.get_compiler(base_path)
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/nikola/post.py", line 138, in __init__
    default_metadata, self.newstylemeta = get_meta(self, self.config['FILE_METADATA_REGEXP'], self.config['UNSLUGIFY_TITLES'])
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/nikola/post.py", line 951, in get_meta
    _, newstylemeta = get_metadata_from_meta_file(post.metadata_path, config, lang)
  File "/Users/lolo/.pyenv/versions/2.7.9/lib/python2.7/site-packages/nikola/post.py", line 903, in get_metadata_from_meta_file
    x.strip() for x in meta_data][:7]
ValueError: need more than 7 values to unpack
````
occurs on MacOsX with homebrew + pyenv + python 2.7.9 or 3.4.3.

Tried to clean-up site and such to no avail.

**Comment by Kwpolska** [comment]
We are aware of this bug (#1689) (it actually hit Nikola’s own site).  It’s caused by old-style metadata used by old posts on your site, combined with a last-minute patch.

To fix, you can just apply the [patch from #1690](https://github.com/getnikola/nikola/pull/1690/files) to your copy of Nikola.  There are also plans to write a metadata converter in #1692.

**Comment by meduz** [comment]
great, I had not spotted this bug.

Since I had only a few old-style metadata files, I converted to the new format. 

Thanks!
